contrib/cirrus/setup_environment.sh

```diff
@@ -192,6 +192,17 @@ EOF
 # on Debian.
 rm -rf /var/lib/containers/storage
 
+# Catch invalid "TMPDIR == /tmp" assumptions; PR #19281
+TMPDIR=$(mktemp --tmpdir -d CI_XXXX)
+# tmp dir is commonly 1777 to allow all user to read/write
+chmod 1777 $TMPDIR
+export TMPDIR
+# Ensure we are actually on tmpfs to speed things up.
+fstype=$(findmnt -n -o FSTYPE --target $TMPDIR)
+if [[ "$fstype" != "tmpfs" ]]; then
+    die "The CI test TMPDIR is not on a tmpfs mount, we need tmpfs to make the tests faster"
+fi
+
```

Comment:
It's fine to do this here (and the related `lib.sh` updates).  I just want to make sure you didn't move it because something I suggested was maybe misinterpreted as a requirement.  I thought what you had previously, with everything nestled in the `runner.sh` function (was it `dotest()`?) was fine.  Only that it might be helped by a comment.  Otherwise assuming this was intentional/required, you can ignore my comment :smile:

Reply:
I mean doing it for all tests is good idea I guess, then again doing it for not all tests is also a good idea (we don't want to become dependant on a set TMPDIR as this is a optional env)

I am going to revert to the old one as I really don't want to change all the lib.sh.t tests to include TMPDIR